### PR TITLE
2419 hook up metrics for undetermined

### PR DESCRIPTION
**Pull request by seallard**
### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/meta/demultiplex/status_db_storage_functions.py [hunk 1]
```diff
@@ -75,10 +70,96 @@ def store_sequencing_metrics_in_status_db(flow_cell: FlowCellDirectoryData, stor
...
+        total_2=new_metric.sample_total_reads_in_lane,
+        percentage_2=new_metric.sample_base_percentage_passing_q30,
+    )
+
+
+def combine_reads(
+    existing_metric: SampleLaneSequencingMetrics, new_metric: SampleLaneSequencingMetrics
+) -> int:
+    """Combine the total reads from two metrics."""
+    return existing_metric.sample_total_reads_in_lane + new_metric.sample_total_reads_in_lane
```
Comment (islean) [inline cg/meta/demultiplex/status_db_storage_functions.py:155]: I personally don't think this needs its own function.

File: cg/meta/demultiplex/status_db_storage_functions.py [hunk 2]
```diff
@@ -75,10 +70,96 @@ def store_sequencing_metrics_in_status_db(flow_cell: FlowCellDirectoryData, stor
...
+
+        existing_metric: SampleLaneSequencingMetrics = (
+            store.get_metrics_entry_by_flow_cell_name_sample_internal_id_and_lane(
+                flow_cell_name=flow_cell.id, sample_internal_id=sample_id, lane=lane
+            )
+        )
+        if existing_metric:
+            update_existing_metric(existing_metric=existing_metric, new_metric=undetermined_metric)
+        else:
+            store.session.add(undetermined_metric)
```
Comment (seallard) [inline cg/meta/demultiplex/status_db_storage_functions.py:105]: Might be necessary to add check whether the associated sample exists in status db before tracking the new metric

File: cg/meta/demultiplex/status_db_storage_functions.py (hunk 1 above)
Comment (seallard) [reply cg/meta/demultiplex/status_db_storage_functions.py:155]: I think it is essential 😠

**Comment by seallard** [comment]
This PR got a bit messy, closed in favour of https://github.com/Clinical-Genomics/cg/pull/2494

**Pull request closed without merging**